### PR TITLE
[SPARK-40947][SPARK-40966][PS][INFRA][TEST] Upgrade pandas to 1.5.1

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes upgrading pandas to 1.5.1, for pandas API on Spark.

New version of pandas (1.5.1) was released last week (Oct 19, 2022).

See [What's new in 1.5.1](https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.5.1.html) for more detail.

### Why are the changes needed?

We should follow the behavior of latest pandas, and support it.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The existing tests should all pass.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
Since we solved the major incompatibility via https://github.com/apache/spark/pull/37955, I'm +1 if CI passed.

**Comment by zhengruifeng** [comment]
seems some tests failed again...

**Comment by itholic** [comment]
Ahh! Let me take a look

**Comment by HyukjinKwon** [comment]
Can you do https://github.com/apache/spark/pull/38437 in this PR together?

**Review by dongjoon-hyun — COMMENTED** [review]
+1 for @HyukjinKwon 's comment.

File: python/pyspark/pandas/tests/test_dataframe_spark_io.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def test_parquet_read_with_pandas_metadata(self):
             expected3 = expected2.set_index("index", append=True)
             # There is a bug in `to_parquet` from pandas 1.5.0 when writing MultiIndex.
             # See https://github.com/pandas-dev/pandas/issues/48848 for the reported issue.
-            if LooseVersion(pd.__version__) == LooseVersion("1.5.0"):
+            if LooseVersion(pd.__version__) > LooseVersion("1.5.0"):
```
Comment (itholic) [inline python/pyspark/pandas/tests/test_dataframe_spark_io.py:102]: Consolidate SPARK-40966 here since the change is small.

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @itholic and @HyukjinKwon .

**Comment by zhengruifeng** [comment]
late lgtm , thanks all

**Pull request closed without merging**